src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeNode.cs

```diff
@@ -704,7 +704,10 @@ protected virtual void ComputeValueTypeFieldPadding()
 
         protected override void OnMarked(NodeFactory context)
         {
-            //Debug.Assert(_type.IsTypeDefinition || !_type.HasSameTypeDefinition(context.ArrayOfTClass), "Asking for Array<T> EEType");
+            if (context is RyuJitNodeFactory)
```

Comment:
I'm not super confident we can pass this assert in that mode. I'll try it, but if it fires I'm not going to address it with this check-in.